bigframes/core/compile/googlesql/expression.py

```diff
@@ -45,8 +47,35 @@ class ColumnExpression(Expression):
 
     def sql(self) -> str:
         if self.parent is not None:
-            return f"{self.parent.sql()}.`{self.name}`"
-        return f"`{self.name}`"
+            return f"{self.parent.sql()}.{identifier(self.name)}"
+        return identifier(self.name)
+
+
+@dataclasses.dataclass
+class ScalarExpression(Expression):
```

Comment:
It looks we can get rid of this expression, if we are able to refactor `vector_search`, `to_query` and `_get_rows_as_json_values` in the block methods. Created work items to follow up: b/347037087, b/347037813, b/347038069